lib/gui/modules/image-writer.js

```diff
@@ -102,12 +113,32 @@ imageWriter.service('ImageWriterService', function($q, $rootScope) {
         flashState.setProgressState(state);
       });
 
-    }).then(flashState.unsetFlashingFlag).catch((error) => {
+    }).then(flashState.unsetFlashingFlag).then(() => {
+      if (flashState.wasLastFlashCancelled()) {
+        AnalyticsService.logEvent('Elevation cancelled', analyticsData);
```

Comment:
@lurch you're assuming that the flash event makes it to mixpanel, but it's not certain. Also, it will make it very difficult to query (basically we won't be able to use mixpanel). As such, whoever is managing this pr, please don't implement the optimisation proposed by Andrew.